packages/agoric-cli/lib/install.js

```diff
@@ -13,15 +13,24 @@ export default async function installMain(progname, rawArgs, powers, opts) {
   const pspawn = makePspawn({ log, spawn, chalk });
 
   const rimraf = file => pspawn('rm', ['-rf', file]);
-  const existingSubdirs = await Promise.all(
-    ['.', '_agstate/agoric-servers', 'contract', 'api', 'ui']
-      .sort()
-      .map(async subd => {
-        const exists = await fs.stat(`${subd}/package.json`).catch(_ => false);
-        return exists && subd;
-      }),
-  );
-  const subdirs = existingSubdirs.filter(subd => subd);
+
+  async function workspaceDirectories(cwd = '.') {
+    // run `yarn workspaces info` to get the list of directories to
+    // use, instead of a hard-coded list
+    const p = pspawn('yarn', ['workspaces', '--silent', 'info'], {
+      cwd,
+      stdio: ['inherit', 'pipe', 'inherit'],
+    });
+    const stdout = [];
+    p.childProcess.stdout.on('data', out => stdout.push(out));
+    await p;
+    const d = JSON.parse(Buffer.concat(stdout).toString('utf-8'));
+    const subdirs = Object.values(d).map(v => v.location);
+    return subdirs;
+  }
+
+  const subdirs = await workspaceDirectories();
```

Comment:
```suggestion
  // Default is for backward compatibility in case the current directory is not a workspace.
  const subdirs = await workspaceDirectories().catch(() => ['api', 'contract', 'ui']);
```

Reply:
I wound up doing something a bit more complicated.. take a look and see what you think.